test/bio_write_test.c

```diff
@@ -0,0 +1,220 @@
+/*
```

Comment:
It looks like much of the code in this file could be combined with the code in bio_read_test. Please try to avoid code duplicaiton.